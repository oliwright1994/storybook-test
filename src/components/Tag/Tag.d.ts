import { MessageType } from '../../constants'

export interface ITag {
  messageType: MessageType
}
