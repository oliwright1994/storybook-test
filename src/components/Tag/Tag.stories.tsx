import { select } from '@storybook/addon-knobs';
import React from 'react';

import createStory from '../../storybook/createStory';
import Tag from '.';
import { MessageType } from '../../constants';

createStory(
  'COMPONENTS | Tag',
  [
    {
      label: 'standard',
      jsx: () => {
        const messageType = select('messageType', MessageType, MessageType.success, 'messageType');
        return <Tag messageType={messageType}>{messageType}</Tag>;
      },
    },
  ],
  { liveEdit: { Component: Tag, render: 'return <Tag messageType="info">hi</Tag>' } }
);
