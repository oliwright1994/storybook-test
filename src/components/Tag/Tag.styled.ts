import styled from 'styled-components';
import { typography, space, color, ColorProps, border, BorderProps, padding, PaddingProps, fontSize, FontSizeProps, fontWeight, FontWeightProps } from 'styled-system';

type TSTag = ColorProps & BorderProps & PaddingProps & FontSizeProps & FontWeightProps

export const STag = styled.div<TSTag>`
  display: inline-block;
  ${fontSize}
  ${padding}
  ${color}
  ${border}
  ${fontWeight}
`;
