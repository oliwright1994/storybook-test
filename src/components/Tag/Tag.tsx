import React, { FC } from 'react'

import { ITag } from './Tag.d'
import { STag } from './Tag.styled'

const Tag: FC<ITag> = ({ children, messageType, ...props }) => (
  <STag
    color={`${messageType}.2`}
    border="2px"
    borderColor={`${messageType}.2`}
    backgroundColor={`${messageType}.0`}
    borderRadius="lg"
    px={2}
    py={1}
    fontSize="xs"
    fontWeight={4}
    {...props}
  >
    {children}
  </STag>
)

export default Tag
