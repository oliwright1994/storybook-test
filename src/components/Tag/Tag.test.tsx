import { render } from '@testing-library/react';
import React from 'react';

import Tag from '.';

describe('<Tag />', () => {
  it('should render as expected', () => {
    const { asFragment } = render(<Tag messageType="success">success</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });
});
