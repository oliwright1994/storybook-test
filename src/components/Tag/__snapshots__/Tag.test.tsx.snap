// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render as expected 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm isQkMb"
    color="success.2"
    font-size="xs"
    font-weight="4"
  >
    success
  </div>
</DocumentFragment>
`;
